src/transformers/modeling_tf_utils.py

```diff
@@ -253,60 +253,49 @@ def detect_tf_missing_unexpected_layers(model, resolved_archive_file):
     unexpected_layers = []
 
     with h5py.File(resolved_archive_file, "r") as f:
-        saved_layer_names = set(hdf5_format.load_attributes_from_hdf5_group(f, "layer_names"))
-        model_layer_names = set(layer.name for layer in model.layers)
-        missing_layers = list(model_layer_names - saved_layer_names)
-        unexpected_layers = list(saved_layer_names - model_layer_names)
+        saved_h5_model_layers_name = set(hdf5_format.load_attributes_from_hdf5_group(f, "layer_names"))
+        model_layers_name_value = {}
 
         for layer in model.layers:
-            if layer.name in saved_layer_names:
-                g = f[layer.name]
-                saved_weight_names = hdf5_format.load_attributes_from_hdf5_group(g, "weight_names")
-                saved_weight_names_set = set(
-                    "/".join(weight_name.split("/")[2:]) for weight_name in saved_weight_names
-                )
```

Comment:
This line and the line defining `symbolic_weights_names` were removed. Instead of being filled by
```py
"/".join(weight_name.split("/")[2:]) for weight_name in saved_weight_names
```
they will now be filled by
```py
"/".join(weight_name.split("/")[1:]) for weight_name in saved_weight_names
```
Why the change from `2:` to `1:`?

Reply:
I took a deeper look and understand the change here. I guess it's because the second index is the the prefix, and since in TF the main layer is named after the prefix, it will remain the same across base models and models with heads.

Reply:
I'm really sorry @LysandreJik I completely forgot to answer :( And yes this is exactly for that :)